test_runner/regress/test_storage_controller.py

```diff
@@ -3093,6 +3093,48 @@ def reconfigure_node_again():
     wait_until(reconfigure_node_again)
 
 
+def test_ps_unavailable_after_delete(neon_env_builder: NeonEnvBuilder):
+    neon_env_builder.num_pageservers = 3
+
+    env = neon_env_builder.init_start()
+
+    def assert_nodes_count(n: int):
+        nodes = env.storage_controller.node_list()
+        assert len(nodes) == n
+
+    # Nodes count must remain the same before deletion
+    assert_nodes_count(3)
+
+    ps = env.pageservers[0]
+    env.storage_controller.node_delete(ps.id)
+
+    # After deletion, the node count must be reduced
+    assert_nodes_count(2)
+
+    log.info("Restarting tombstoned pageserver...")
+    ps.restart()
```

Comment:
This will hang since `NeonPageserver.start` waits for the node to be marked as active in the storage controller by default. Re-attaching is a pre-condition for that. To avoid this, you can do:
```
ps.stop()
ps.start(await_active=False)
```